### PR TITLE
feat(macOS): Implement tray icon template

### PR DESCRIPTION
**Pull request by lemarier**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [ ] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by lemarier** [comment]
Should be merged AFTER https://github.com/tauri-apps/tauri/pull/2321

**Comment by lucasfernog** [comment]
I was just finishing this too :P gonna see how it looks.

**Comment by lemarier** [comment]
> I was just finishing this too :P gonna see how it looks.

feel free to redoit

not sure it was the right way

File: core/tauri-utils/src/config.rs [hunk 1]
```diff
@@ -209,6 +209,9 @@ pub struct SystemTrayConfig {
   /// Path to the icon to use on the system tray.
   /// Automatically set to be an `.png` on macOS and Linux, and `.ico` on Windows.
   pub icon_path: PathBuf,
+  /// A Boolean value that determines whether the image represents a [template](https://developer.apple.com/documentation/appkit/nsimage/1520017-template?language=objc) image on macOS.
+  #[serde(default)]
+  pub icon_as_template: bool,
```
Comment (lucasfernog) [inline core/tauri-utils/src/config.rs:214]: also need to add this to the config_definition code on cli.rs

**Review by lucasfernog — APPROVED** [review]
LGTM

**Pull request merged**